fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/logical/types/LogicalTypesTestBase.java

```diff
@@ -0,0 +1,220 @@
+package com.linkedin.avro.fastserde.logical.types;
+
+import static com.linkedin.avro.fastserde.FastSerdeTestsSupport.getCodeGenDirectory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.avro.Conversion;
+import org.apache.avro.Conversions;
+import org.apache.avro.Schema;
+import org.apache.avro.data.TimeConversions;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificData;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.testng.Assert;
+import org.testng.collections.Lists;
+
+import com.linkedin.avro.fastserde.FastDeserializer;
+import com.linkedin.avro.fastserde.FastGenericDeserializerGenerator;
+import com.linkedin.avro.fastserde.FastGenericSerializerGenerator;
+import com.linkedin.avro.fastserde.FastSerializer;
+import com.linkedin.avro.fastserde.FastSpecificDeserializerGenerator;
+import com.linkedin.avro.fastserde.FastSpecificSerializerGenerator;
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelperCommon;
+import com.linkedin.avroutil1.compatibility.AvroVersion;
+import com.linkedin.avroutil1.compatibility.SchemaNormalization;
+
+public abstract class LogicalTypesTestBase {
+
+    private final int v1HeaderLength = 10;
+    private final File classesDir;
+    private final ClassLoader classLoader;
+
+    LogicalTypesTestBase() {
+        try {
+            classesDir = getCodeGenDirectory();
+            classLoader = URLClassLoader.newInstance(new URL[]{classesDir.toURI().toURL()}, getClass().getClassLoader());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected <T extends GenericData> T copyConversions(SpecificData fromSpecificData, T toModelData) {
+        Optional.ofNullable(fromSpecificData.getConversions())
+                .orElse(Collections.emptyList())
+                .forEach(toModelData::addLogicalTypeConversion);
+
+        fixConversionsIfAvro19(toModelData);
+
+        return toModelData;
+    }
+
+    protected <T extends GenericData> void fixConversionsIfAvro19(T modelData) {
```

Comment:
I don't get that - if the conversions are missing by default in Avro 1.9 then why we are forcefully adding them?

Reply:
Conversions not missing. They're generated incompletely. **Avro classes are generated with logical-types** (e.g. LocalTime) however **some of conversions are missing.**
The bug is in avro-compiler module. 

I created an example which shows that:
- https://github.com/krisso-rtb/avro-logical-types-generated-classes/pull/1/files

Search for `missingConversions`. It's a hello-world-level project so it should be easy to run it locally.